### PR TITLE
feat: add poetry.lock file and update the files with pyansys guidelines

### LINKED ISSUE #174: [MAINTENANCE] Missing or outdated CONTRIBUTORS.md file in ansys/pytwin
Please add the CONTRIBUTORS.md file to the repository. The file might either be missing or is an outdated version of the expected file.

See [here](https://github.com/ansys-internal/pyansys-maintenance-automation/blob/main/scripts/file_references/CONTRIBUTORS.md) an up-to-date reference to a properly formatted file.

---

This issue was created by a bot. If you have any questions, please ping the PyAnsys Core team for assistance. To ping the team write a comment in this issue with the following mention: `@ansys/pyansys-core`.

If you want this repository to be excluded from this automated maintenance process, please let us know by [filling in the opt-out request form](https://github.com/ansys-internal/pyansys-maintenance-automation/issues/new?labels=&projects=&template=opt-out.yml&title=Request+to+opt-out+from+automated+maintenance+for+ansys%2Fpytwin).

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
closes #174 #173

### CODE REVIEW

**Review by lboucin — COMMENTED** [review]
LGTM ! Thanks for the reat maintenance work here @Revathyvenugopal162.
Also, I think some maintenance will have to be done on our windows VM to have the pipeline fully successful.
@chrpetre that is maybe the right time to do the move to a github hosted agent for windows related actions (I guess doc build is the last one still running on the VM right?)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -99,7 +99,7 @@ jobs:
     strategy:
        matrix:
            os: [windows-latest, ubuntu-20.04]
-           python-version: ["3.9", "3.10", "3.11", "3.12"]
+           python-version: ["3.10", "3.11", "3.12"]
```
Comment (Revathyvenugopal162) [inline .github/workflows/ci.yml:102]:
```suggestion
           python-version: ["3.9", "3.10", "3.11", "3.12"]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -112,7 +114,7 @@ jobs:
         uses: pyvista/setup-headless-display-action@v2
 
       - name: Run pytest
-        uses: ansys/actions/tests-pytest@v7
+        uses: ansys/actions/tests-pytest@fix/poetry-tests-pytest
```
Comment (Revathyvenugopal162) [inline .github/workflows/ci.yml:117]:
```suggestion
        uses: ansys/actions/tests-pytest@v8
```

File: pyproject.toml [hunk 3]
```diff
@@ -23,13 +23,39 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.9,<4"
-importlib-metadata = {version = "^4.0", python = "<3.8"}
+python = ">=3.9,<3.13"
```
Comment (RobPasMue) [inline pyproject.toml:26]:
```suggestion
python = ">=3.9,<4"
```
Maybe better like this?

File: tox.ini [hunk 4]
```diff
@@ -1,31 +1,30 @@
...
+    3.11: style,py311-coverage,doc
+    3.12: style,py312-coverage,doc
 [testenv]
+allowlist_externals = poetry
 description = Checks for project unit tests and coverage (if desired)
 basepython =
-    py37: python3.7
-    py38: python3.8
-    py39: python3.9
     py310: python3.10
```
Comment (RobPasMue) [inline tox.ini:19]:
```suggestion
    py39: python3.9
    py310: python3.10
```

**Review by RobPasMue — COMMENTED** [review]
LGTM!

File: pyproject.toml (hunk 3 above)
Comment (Revathyvenugopal162) [reply pyproject.toml:26]: image
There was a conflict when updating to version 4, so I had to limit it to version 3.13.

**Review by lboucin — APPROVED** [review]
LGTM! Thank you for the great work @Revathyvenugopal162!

**Comment by chrpetre** [comment]
thanks @Revathyvenugopal162 for all the updates here!

**Pull request merged**